src/main/scala/mesosphere/marathon/core/check/Check.scala

```diff
@@ -0,0 +1,245 @@
+package mesosphere.marathon
+package core.check
+
+import com.wix.accord._
+
+import mesosphere.marathon.Protos.CheckDefinition.Protocol
+import mesosphere.marathon.state._
+import org.apache.mesos.{Protos => MesosProtos}
+
+import scala.concurrent.duration._
+
+sealed trait Check {
+  def delay: FiniteDuration
+  def interval: FiniteDuration
+  def timeout: FiniteDuration
+  def toProto: Protos.CheckDefinition
+
+  protected def protoBuilder: Protos.CheckDefinition.Builder =
```

Comment:
Is the the storage proto or Mesos proto?

Reply:
storage